fix(language_server): fix panic when "disable rule for this line" position is after error span

Sysix opened the pull request:
The problem is, that `get_section_insert_position` get the offset after the line break.
But the error offset is before the insert offset. So it gets a range of `[4..3]`. 

This rule makes no sense to ignore, because the span will always be 0 with a length of 0. 
Ignored the case for adding "ignore fixes" and fixed the panic to make sure, it will not happen for other rules.

closes #14565

camc314 commented:
### Merge activity

* **Oct 15, 9:03 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 15, 9:03 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).
* **Oct 15, 9:07 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
fix(language_server): fix panic when "disable rule for this line" position is after error span (#14597)

The problem is, that `get_section_insert_position` get the offset after the line break.
But the error offset is before the insert offset. So it gets a range of `[4..3]`.

This rule makes no sense to ignore, because the span will always be 0 with a length of 0.
Ignored the case for adding "ignore fixes" and fixed the panic to make sure, it will not happen for other rules.

closes #14565
